src/components/settings/PushNotifications/GlobalPushNotifications.tsx

```diff
@@ -93,22 +94,30 @@ export const _sanitizeNotifiableSafes = (
   }, {})
 }
 
-// Merges added Safes and currently notified Safes into a single data structure without duplicates
+// Merges added Safes, currently notified Safes, and owned safes into a single data structure without duplicates
 export const _mergeNotifiableSafes = (
+  ownedSafes: AllOwnedSafes | undefined,
   addedSafes: AddedSafesState,
   currentSubscriptions?: NotifiableSafes,
-): NotifiableSafes => {
-  const notifiableSafes = _transformAddedSafes(addedSafes)
-
-  if (!currentSubscriptions) {
-    return notifiableSafes
-  }
-
-  // Locally registered Safes (if not already added)
-  for (const [chainId, safeAddresses] of Object.entries(currentSubscriptions)) {
-    const notifiableSafesOnChain = notifiableSafes[chainId] ?? []
-    const uniqueSafeAddresses = Array.from(new Set([...notifiableSafesOnChain, ...safeAddresses]))
+): NotifiableSafes | undefined => {
+  const added = _transformAddedSafes(addedSafes)
+
+  const chains = Array.from(
+    new Set([
+      ...Object.keys(addedSafes || {}),
+      ...Object.keys(currentSubscriptions || {}),
+      ...Object.keys(ownedSafes || {}),
+    ]),
+  )
 
+  let notifiableSafes: NotifiableSafes = {}
+  for (const chainId of chains) {
+    const ownedSafesOnChain = ownedSafes?.[chainId] ?? []
+    const addedSafesOnChain = added[chainId] || []
+    const currentSubscriptionsOnChain = currentSubscriptions?.[chainId] || []
+    const uniqueSafeAddresses = Array.from(
+      new Set([...currentSubscriptionsOnChain, ...addedSafesOnChain, ...ownedSafesOnChain]),
```

Comment:
I think we need to filter `addedSafesOnChain` and remove all safes that are not owned otherwise the user can see un-owned safes that are in their watchlist on the global preferences page.

Reply:
I assumed there was some reason for allowing to subscribe to notifications even if they don't own the safe but I guess there wouldn't be use for this. So yeah I can just show owned